test/test_transforms_v2_refactored.py

```diff
@@ -1634,3 +1639,111 @@ def test_transform_negative_degrees_error(self):
+            check_dispatch=False,  # TODO: the check would pass if we were to use the non-datapoint dependent logic of _check_dispatcher_dispatch ¯\_(ツ)_/¯
+            dtype=output_dtype,
+            scale=scale,
+        )
+
+    @pytest.mark.parametrize(
+        "make_input",
+        [make_image_tensor, make_image, make_bounding_box, make_segmentation_mask, make_video],
+    )
+    @pytest.mark.parametrize("input_dtype", [torch.float32, torch.float64, torch.uint8])
+    @pytest.mark.parametrize("output_dtype", [torch.float32, torch.float64, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("scale", (True, False))
+    def test_transform(self, make_input, input_dtype, output_dtype, device, scale):
+        input = make_input(dtype=input_dtype, device=device)
+        output_dtype = {"others": torch.float32}
+
+        check_transform(transforms.ToDtype, input, dtype=output_dtype, scale=scale)
+
+    def reference_convert_dtype_image_tensor(self, image, dtype=torch.float, scale=False):
+        input_dtype = image.dtype
+        output_dtype = dtype
+
+        if not scale:
+            return image.to(dtype)
+
+        if output_dtype == input_dtype:
+            return image
+
+        def fn(value):
+            if input_dtype.is_floating_point:
+                if output_dtype.is_floating_point:
+                    return value
+                else:
+                    return int(decimal.Decimal(value) * torch.iinfo(output_dtype).max)
+            else:
+                input_max_value = torch.iinfo(input_dtype).max
+
+                if output_dtype.is_floating_point:
+                    return float(decimal.Decimal(value) / input_max_value)
+                else:
+                    output_max_value = torch.iinfo(output_dtype).max
+
+                    if input_max_value > output_max_value:
+                        factor = (input_max_value + 1) // (output_max_value + 1)
+                        return value / factor
+                    else:
+                        factor = (output_max_value + 1) // (input_max_value + 1)
+                        return value * factor
+
+        return torch.tensor(tree_map(fn, image.tolist()), dtype=dtype)
+
+    @pytest.mark.parametrize("input_dtype", [torch.float32, torch.float64, torch.uint8])
+    @pytest.mark.parametrize("output_dtype", [torch.float32, torch.float64, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("scale", (True, False))
+    def test_against_ref(self, input_dtype, output_dtype, device, scale):
+        input = make_image(dtype=input_dtype, device=device)
+
+        out = F.to_dtype(input, dtype=output_dtype, scale=scale)
+        expected = self.reference_convert_dtype_image_tensor(input, dtype=output_dtype, scale=scale)
+
+        if output_dtype is torch.uint8 and scale:
+            # TODO: is this actually normal? Why wasn't this a problem before?
```

Comment:
@pmeier

Reply:
We don't have any tolerances set in the legacy v2 tests. However, we aren't testing full images there, but rather only

https://github.com/pytorch/vision/blob/cc0f9d02e1e7549e9e8ef1081a0d3b5d72f1297d/test/transforms_v2_kernel_infos.py#L1734-L1739

I'll have a look at the reference and get back to you.

Reply:
Ok, I can reproduce the issue when using proper images in the legacy tests. The error we are seeing stems from

https://github.com/pytorch/vision/blob/cc0f9d02e1e7549e9e8ef1081a0d3b5d72f1297d/torchvision/transforms/v2/functional/_meta.py#L321-L332

Two issues here:

1. Our reference is wrong.
   https://github.com/pytorch/vision/blob/cc0f9d02e1e7549e9e8ef1081a0d3b5d72f1297d/test/transforms_v2_kernel_infos.py#L1696
   It should be `round` rather than `int` following the comment above.
2. Albeit fewer, even with 1. fixed I'm still seeing some off-by-one errors. I've checked and this just stems from the fact that the more performant strategy we are using in the kernel, can produce slightly different results than the exact one we have as reference.

So an `atol=1` is fine in case we transform from float to int.